opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -141,25 +141,36 @@
 class Resource:
     """A Resource is an immutable representation of the entity producing telemetry as Attributes."""
 
-    def __init__(self, attributes: Attributes):
+    def __init__(
+        self, attributes: Attributes, schema_url: typing.Optional[str] = None
+    ):
         _filter_attributes(attributes)
         self._attributes = attributes.copy()
+        if schema_url is None:
+            schema_url = ""
+        self._schema_url = schema_url
 
     @staticmethod
-    def create(attributes: typing.Optional[Attributes] = None) -> "Resource":
+    def create(
+        attributes: typing.Optional[Attributes] = None,
+        schema_url: typing.Optional[str] = None,
+    ) -> "Resource":
         """Creates a new `Resource` from attributes.
 
         Args:
             attributes: Optional zero or more key-value pairs.
+            schema_url: Optional URL pointing to the schema
 
         Returns:
             The newly-created Resource.
         """
         if not attributes:
             attributes = {}
+        if not schema_url:
```

Comment:
is this needed here if the Resource constructor does the same check?

Reply:
It is not. It's a holdover from when I was concerned about the constructor and factory method behaving differently. I'll remove it from the factory method, but I wonder if it'd be better to remove it from the constructor instead, as directly instantiating Resource objects isn't supported.

Reply:
I think either is fine 👍